receiver/azureeventhubreceiver/azureresourcemetrics_unmarshaler.go

```diff
@@ -0,0 +1,126 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package azureeventhubreceiver // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/azureeventhubreceiver"
+
+import (
+	"bytes"
+	"time"
+
+	eventhub "github.com/Azure/azure-event-hubs-go/v3"
+	jsoniter "github.com/json-iterator/go"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+	conventions "go.opentelemetry.io/collector/semconv/v1.13.0"
+
+	"go.uber.org/zap"
+)
+
+type azureResourceMetricsUnmarshaler struct {
+	buildInfo component.BuildInfo
+	logger    *zap.Logger
+}
+
+// azureMetricRecords represents an array of Azure metric records
+// as exported via an Azure Event Hub
+type azureMetricRecords struct {
+	Records []azureMetricRecord `json:"records"`
+}
+
+// azureMetricRecord represents a single Azure Metric following
+// the common schema does not exist (yet):
+type azureMetricRecord struct {
+	Time       string  `json:"time"`
+	ResourceID string  `json:"resourceId"`
+	MetricName string  `json:"metricName"`
+	TimeGrain  string  `json:"timeGrain"`
+	Total      float64 `json:"total"`
+	Count      float64 `json:"count"`
+	Minimum    float64 `json:"minimum"`
+	Maximum    float64 `json:"maximum"`
+	Average    float64 `json:"average"`
+}
+
+func newAzureResourceMetricsUnmarshaler(buildInfo component.BuildInfo, logger *zap.Logger) eventhubMetricsUnmarshaller {
+
+	return azureResourceMetricsUnmarshaler{
+		buildInfo: buildInfo,
+		logger:    logger,
+	}
+}
+
+// UnmarshalMetrics takes a byte array containing a JSON-encoded
+// payload with Azure metric records and transforms it into
+// an OpenTelemetry pmetric.Metrics object. The data in the Azure
+// metric record appears as fields and attributes in the
+// OpenTelemetry representation;
+func (r azureResourceMetricsUnmarshaler) UnmarshalMetrics(event *eventhub.Event) (pmetric.Metrics, error) {
+
+	md := pmetric.NewMetrics()
+
+	var azureMetrics azureMetricRecords
+	decoder := jsoniter.NewDecoder(bytes.NewReader(event.Data))
+	err := decoder.Decode(&azureMetrics)
+	if err != nil {
+		return md, err
+	}
+
+	resourceMetrics := md.ResourceMetrics().AppendEmpty()
+	scopeMetrics := resourceMetrics.ScopeMetrics().AppendEmpty()
+
+	scopeMetrics.Scope().SetName(receiverScopeName)
+	scopeMetrics.Scope().SetVersion(r.buildInfo.Version)
```

Comment:
could this go at the resource level?

Reply:
The only difference is economical: the attributes apply anyway, it’s just that the resource is more generic.

Reply:
I could use some help in getting this change in place. I looked for some examples but couldn't determine how to set the values on the Resource unless I use Attributes.

Reply:
Sorry, I am misleading you on this one. You are correct that name and version are set at the scope level.